functions-and-operators/string-functions.md

```diff
@@ -179,11 +179,198 @@ Return the length of a string in bytes.
+
+SELECT '🍣🍺sushi🍣🍺' LIKE '%sushi%' AS result;
++--------+
+| result |
++--------+
+|      1 |
++--------+
+
+SELECT '🍣🍺sushi🍣🍺' LIKE '%🍣%' AS result;
++--------+
+| result |
++--------+
+|      1 |
++--------+
+```
+
+If you don't specify the `ESCAPE` character, `\` will be assumed.
+```sql
+SELECT 'sushi!!!' LIKE 'sushi\_' AS result;
++--------+
+| result |
++--------+
+|      0 |
++--------+
+
+SELECT 'sushi_' LIKE 'sushi\_' AS result;
++--------+
+| result |
++--------+
+|      1 |
++--------+
+```
+
+To specify the different escape character, you can use `ESCAPE` clause.
+```sql
+SELECT 'sushi_' LIKE 'sushi*_' ESCAPE '*' AS result;
++--------+
+| result |
++--------+
+|      1 |
++--------+
+
+SELECT 'sushi!' LIKE 'sushi*_' ESCAPE '*' AS result;
++--------+
+| result |
++--------+
+|      0 |
++--------+
+```
+
+`LIKE` permits numeric expressions.
+```sql
+SELECT 10 LIKE '1%' AS result;
++--------+
+| result |
++--------+
+|      1 |
++--------+
+
+SELECT 10000 LIKE '12%' AS result;
++--------+
+| result |
++--------+
+|      0 |
++--------+
+```
 
 ### [`LOCATE()`](https://dev.mysql.com/doc/refman/8.0/en/string-functions.html#function_locate)
 
-Return the position of the first occurrence of substring.
+`LOCATE(substr, str[, pos])` returns the position of first occurrence of substring in string.
+When you specify `pos`, `LOCATE()` starts seeking at the position `pos`.
+`LOCATE` returns `0` if `substr` not in `str` and returns `NULL` if any arguments are null.
+This function is multi-byte safe and is case-sensitive only if at least one argument is a binary string.
+```sql
+SELECT LOCATE('bar', 'foobarbar');
++----------------------------+
+| LOCATE('bar', 'foobarbar') |
++----------------------------+
+|                          4 |
++----------------------------+
+
+SELECT LOCATE('baz', 'foobarbar');
++----------------------------+
+| LOCATE('baz', 'foobarbar') |
++----------------------------+
+|                          0 |
++----------------------------+
+
+SELECT LOCATE('bar', 'fooBARBAR');
++----------------------------+
+| LOCATE('bar', 'fooBARBAR') |
++----------------------------+
+|                          4 |
++----------------------------+
+
+SELECT LOCATE('bar', 'fooBARBAR', 100);
++---------------------------------+
+| LOCATE('bar', 'fooBARBAR', 100) |
++---------------------------------+
+|                               0 |
++---------------------------------+
+
+SELECT LOCATE('bar', 'fooBARBAR', 5);
++-------------------------------+
+| LOCATE('bar', 'fooBARBAR', 5) |
++-------------------------------+
+|                             7 |
++-------------------------------+
+
+SELECT LOCATE('bar', NULL);
++---------------------+
+| LOCATE('bar', NULL) |
++---------------------+
+|                NULL |
++---------------------+
+
+SELECT LOCATE('い', 'たいでぃーびー');
++----------------------------------------+
+| LOCATE('い', 'たいでぃーびー')         |
++----------------------------------------+
+|                                      2 |
```

Comment:
I'm getting different values.
```
mysql> SELECT LOCATE('い', 'たいでぃーびー', 3);
+-------------------------------------------+
| LOCATE('い', 'たいでぃーびー', 3)         |
+-------------------------------------------+
|                                         4 |
+-------------------------------------------+
1 row in set (0.00 sec)
```

Reply:
I think it may be the difference of collation, so I mentioned using `utf8mb4_bin` and specify explicitly with other collation
e.g. https://github.com/pingcap/docs/pull/16084/commits/4a9572539ed773ea0d8e2b6faa40463b0a49bdce#diff-ca49915f7ccf3346527f215e50d580ad5df1526e72f8c7cdcde992500384239aR379-R393